pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/phimanifest/PhiManifestService.java

```diff
@@ -0,0 +1,267 @@
+
+@Slf4j
+public class PhiManifestService {
+    private static final String CONSENT_ADDENDUM_PEDIATRICS_ACTIVITY_STABLE_ID = "CONSENT_ADDENDUM_PEDIATRIC";
+    private static final String SOMATIC_CONSENT_TUMOR_PEDIATRIC_QUESTION = "SOMATIC_CONSENT_TUMOR_PEDIATRIC";
+    private static final String SOMATIC_ASSENT_ADDENDUM_QUESTION = "SOMATIC_ASSENT_ADDENDUM";
+    private static final String CONSENT_ADDENDUM_ACTIVITY_STABLE_ID = "CONSENT_ADDENDUM";
+    private static final String LMS_QUESTION_SOMATIC_CONSENT_ADDENDUM_TUMOR = "SOMATIC_CONSENT_ADDENDUM_TUMOR";
+    private static final String OS2_QUESTION_SOMATIC_CONSENT_ADDENDUM_TUMOR = "SOMATIC_CONSENT_TUMOR";
+
+    private final String[] headers = new String[] {"Short Id", "First Name", "Last Name", "Proxy First Name", "Proxy Last Name",
+            "Date of Birth", "Gender", "Somatic Assent Addendum Response", "Somatic Consent Tumor Pediatric Response",
+            "somatic Consent Tumor Response", "Date of PX", "Facility Name", "Sample Type", "Accession Number", "Histology",
+            "Block Id", "Tumor Collaborator Sample Id", "Tissue Site", "Sequencing Results", "Normal Manufacturer Barcode",
+            "Normal Collaborator Sample Id", "Clinical Order Date", "Clinical Order Id", "Clinical PDO Number", "Order Status",
+            "Order Status Date"};
+
+
+    public PhiManifestResponse generateReport(String ddpParticipantId, String sequencingOrderId, DDPInstanceDto ddpInstanceDto) {
+        MercuryOrderDao mercuryOrderDao = new MercuryOrderDao();
+        List<MercuryOrderDto> orders = mercuryOrderDao.getByOrderId(sequencingOrderId);
+        if (!mercuryOrderDao.isSequencingOrderValidForPhiReport(orders, ddpParticipantId, ddpInstanceDto)) {
+            String errorMessage = String.format("Sequencing order number %s is not a valid order for participant", sequencingOrderId);
+            log.warn(errorMessage);
+            return new PhiManifestResponse(errorMessage);
+        }
+        if (!isParticipantConsented(ddpParticipantId, ddpInstanceDto)) {
+            String errorMessage = String.format("Participant %s has not consented to receive shared learning ", ddpParticipantId);
+            log.warn(errorMessage);
+            return new PhiManifestResponse(errorMessage);
+        }
+        ElasticSearchParticipantDto participant = ElasticSearchUtil.getParticipantESDataByParticipantId(
+                ddpInstanceDto.getEsParticipantIndex(), ddpParticipantId);
+        if (participant.getProfile().isEmpty()) {
+            throw new DsmInternalError("There is no profile in ES for participant " + ddpParticipantId);
+        }
+        PhiManifest finalPhiManifest = generateDataForReport(participant, orders, ddpInstanceDto);
+        return createResponse(finalPhiManifest, ddpParticipantId, sequencingOrderId);
+    }
+
+    private PhiManifestResponse createResponse(PhiManifest finalPhiManifest, String ddpParticipantId, String sequencingOrderId) {
```

Comment:
This can move to a method in `PhiManifest`, perhaps in a `toDataArray()` method.